rust/src/transformations/quantile_score_candidates/score_candidates_map.tex

```diff
@@ -0,0 +1,179 @@
+            \\&\hspace{1cm} - (|(1 - \alpha) \cdot \#(x < C_i) - \alpha \cdot \#(x > C_i)| + |\pm(1 - \alpha)|)| &&\text{by triangle inequality} \\
+        &= \alpha_{den} \cdot \max\limits_{i} |1 - \alpha| &&\text{scores cancel}\\
+        &= \alpha_{den} - \alpha_{num} &&\text{since } \alpha \leq 1
+    \intertext{Case 2. Assume $x'$ is equal to $x$, but with some $x_j > C_i$ added or removed.}
+        &= \alpha_{den} \cdot \max\limits_{i} ||(1 - \alpha) \cdot \#(x < C_i) - \alpha \cdot \#(x > C_i)| 
+            \\&\hspace{1cm} - |(1 - \alpha) \cdot \#(x < C_i) - \alpha \cdot (\#(x > C_i) \pm 1)|| \\
+        &\leq \alpha_{den} \cdot \max\limits_{i} ||(1 - \alpha) \cdot \#(x < C_i) - \alpha \cdot \#(x > C_i)| 
+            \\&\hspace{1cm} - (|(1 - \alpha) \cdot \#(x < C_i) - \alpha \cdot \#(x > C_i)| + |\pm\alpha|)| &&\text{by triangle inequality} \\
+        &= \alpha_{den} \cdot \max\limits_{i} |\alpha| &&\text{scores cancel}\\
+        &= \alpha_{num} &&\text{since } \alpha \geq 0
+    \intertext{Case 3. Assume $x'$ is equal to $x$, but with some $x_j = C_i$ added or removed.}
+        &= \alpha_{den} \cdot \max\limits_{i} ||(1 - \alpha) \cdot \#(x < C_i) - \alpha \cdot \#(x > C_i)| 
+            \\&\hspace{1cm} - |(1 - \alpha) \cdot \#(x < C_i) - \alpha \cdot \#(x > C_i)|| \\
+        &= 0 &&\text{no change in score}
+    \intertext{Take the union bound over all cases.}
+        &\le \max(\alpha_{num}, \alpha_{den} - \alpha_{num})
+    \end{align*}
+
+\end{proof}
+
+\label{sec:known-size}
+Now consider the case where the dataset size is known.
+
+\begin{lemma}
+    \label{single-change-stab}
+    If $d_{CO}(x, x') \le 1$, then $d_\infty(\function(x), \function(x')) \le 2 \cdot \alpha_{den}$.
+\end{lemma}
+
+\begin{proof}
+    Assume $d_{CO}(x, x') \le 1$. 
+
+    \begin{align*}
+        &d_\infty(\function(x), \function(x')) \\
+        &= \max\limits_{i} |\function(x)_i - \function(x')_i| &&\text{by definition of } d_\infty \\
+        &= \max\limits_{i} |\mathrm{abs\_diff}(\alpha_{den} \cdot \min(\#(x < C_i), l), \alpha_{num} \cdot \min(|x| - \#(x = C_i), l)) &&\text{by def. of } \function
+            \\&\hspace{1cm} -\mathrm{abs\_diff}(\alpha_{den} \cdot \min(\#(x' < C_i), l), \alpha_{num} \cdot \min(|x'| - \#(x' = C_i), l))| \\
+        &= \alpha_{den} \cdot \max\limits_{i} ||\min(\#(x < C_i), l) - \alpha \cdot \min(|x| - \#(x = C_i), l)|
+            \\&\hspace{1cm}-|\min(\#(x' < C_i), l) - \alpha \cdot \min(|x'| - \#(x' = C_i), l)|| \\
+        &= \alpha_{den} \cdot \max\limits_{i} ||\#(x < C_i) - \alpha \cdot (|x| - \#(x = C_i))|
+            \\&\hspace{1cm}-|\#(x' < C_i) - \alpha \cdot (|x| - \#(x' = C_i))|| \\
+    \end{align*}
+
+    \begin{align*}
+    \intertext{Consider each of the four cases of changing a row in x.}
+    \intertext{Case 1. Assume $x'$ is equal to $x$, but with some $x_j < C_i$ replaced with $x'_j > C_i$.}
```

Comment:
This is where an extra two came from in the prior proof. In the prior proof, it considered the addition and removal separately.